base/abstractarray.jl

```diff
@@ -2130,18 +2166,28 @@ end
     foreach(f, c...) -> Nothing
 
 Call function `f` on each element of iterable `c`.
-For multiple iterable arguments, `f` is called elementwise.
-`foreach` should be used instead of `map` when the results of `f` are not
+For multiple iterable arguments, `f` is called elementwise, and iteration stops when
+any iterator is finished.
+
+`foreach` should be used instead of [`map`](@ref) when the results of `f` are not
 needed, for example in `foreach(println, array)`.
 
 # Examples
 ```jldoctest
-julia> a = 1:3:7;
+julia> tri = 1:3:7; res = Int[];
 
-julia> foreach(x -> println(x^2), a)
-1
-16
-49
+julia> foreach(x -> push!(res, x^2), tri)
+
+julia> res
+3-element Vector{$(Int)}:
+  1
+ 16
+ 49
+
+julia> foreach((x,y) -> println(x," & ",y), tri, 1:100)
```

Comment:
```suggestion
julia> foreach((x, y) -> println(x, " & ", y), tri, 1:100)
```

Reply:
Thanks for taking on the merge, I meant to have a go... 

Here, it's possible that writing something which is more obviously not code would be better, maybe `" and "`